scripts/fix-engine-names.py

```diff
@@ -86,16 +86,18 @@ def addFixedStockfish15():
 
 def amendEngineNames(player):
     name, space, rest = player.partition(" ")
+    # "Cheng 4 0.36c" will then be dealt with in fixVariousVersions() below
```

Comment:
As said in the earlier commit, I think better would be to just add all these in the big dictionary. The we don't have to do multistage substitutions, which is inherently error-prone.

There are not that many of these exceptional names.